server-common/src/main/java/org/apache/kafka/server/mutable/BoundedList.java

```diff
@@ -35,24 +35,12 @@ public class BoundedList<E> implements List<E> {
     private final int maxLength;
     private final List<E> underlying;
 
-    public static <E> BoundedList<E> newArrayBacked(int maxLength) {
-        return new BoundedList<>(maxLength, new ArrayList<>());
-    }
-
-    public static <E> BoundedList<E> newArrayBacked(int maxLength, int initialCapacity) {
-        return new BoundedList<>(maxLength, new ArrayList<>(initialCapacity));
-    }
-
-    public BoundedList(int maxLength, List<E> underlying) {
+    public BoundedList(int maxLength) {
         if (maxLength <= 0) {
             throw new IllegalArgumentException("Invalid non-positive maxLength of " + maxLength);
         }
         this.maxLength = maxLength;
-        if (underlying.size() > maxLength) {
-            throw new BoundedListTooLongException("Cannot wrap list, because it is longer than " +
-                "the maximum length " + maxLength);
-        }
-        this.underlying = underlying;
+        this.underlying = new ArrayList<>(maxLength);
```

Comment:
It seems to me the large batch operation is not usual. It could be a potential performance regression if we always create `ArrayList` with `10000` initial size.

If the origin constructor has potential issue about using a reference to `underlying`, we can change it from `public` to `private`.

Reply:
Sorry I am confused on what you want to change to `private`? Today, passing in the source list is only used in unit tests, not any source code. If it's only used for unit tests I think it should be removed so that tests interact with the BoundedList the same way the source code does.

Reply:
Yeah, it seems like we want to make the constructor private and use the newArrayBacked methods to accomplish what we want. 

I think we can all agree directly passing in the underlying arraylist is not what we want 👍

Reply:
Got it, so we can keep the existing `newArrayBacked()` constructors and get rid of the constructor that takes in a list.